scripts/prep_data_for_model.R

```diff
@@ -13,25 +53,104 @@ unique(standard_catch$subsite)
+                                  T ~ NA)) |> 
+  select(-species, -release_id, -is_yearling, -month, -day, -cutoff) |> 
+  glimpse()
+
+
+# add logic to assign lifestage_for_model 
+# extrapolate lifestage for model for plus count fish/fish without fork lenghts based on weekly fl probabilities
+# Create table with prob fry, smolt, and yearlings for each stream, site, week, year
+weekly_lifestage_bins <- standard_catch_unmarked |> 
+  filter(!is.na(fork_length), count != 0) |> 
+  mutate(year = year(date), week = week(date)) |> 
+  group_by(year, week, stream, site) |> 
+  summarize(percent_fry = sum(lifestage_for_model == "fry")/n(),
+            percent_smolt = sum(lifestage_for_model == "smolt")/n(),
+            percent_yearling = sum(lifestage_for_model == "yearling")/n()) |> 
+  ungroup() |> 
+  glimpse() 
+
+# create table of all na values that need to be filled
+# 39,141 rows 
+na_filled_lifestage <- standard_catch_unmarked |> 
+  mutate(week = week(date), year = year(date)) |> 
+  filter(is.na(fork_length) & count > 0) |> 
+  left_join(weekly_lifestage_bins) |> 
+  mutate(fry = round(count * percent_fry), 
+         smolt = round(count * percent_smolt), 
+         yearling = round(count * percent_yearling)) |> 
+  select(-lifestage_for_model, -count, -week, -year) |> # remove because all na, assigning in next line
+  pivot_longer(fry:yearling, names_to = 'lifestage_for_model', values_to = 'count') |> 
+  select(-c(percent_fry, percent_smolt, percent_yearling)) |>  
+  filter(count != 0) |> # remove 0 values introduced when 0 prop of a lifestage, significantly decreases size of DF 
+  mutate(model_lifestage_method = "assign count based on weekly distribution") |> 
+  glimpse()
+
+# add filled values back into combined_rst 
+# first filter combined rst to exclude rows in na_to_fill
+# total of 
+combined_rst_wo_na_fl <- standard_catch_unmarked |>   
+  filter(!is.na(fork_length)) |> 
+  mutate(model_lifestage_method = "assigned from fl cutoffs") |> 
+  glimpse()
+
+no_catch <- standard_catch_unmarked |> 
+  filter(is.na(fork_length) & count == 0)
+
+# less rows now than in origional, has to do with removing count != 0 in line 104, is there any reason not to do this?
+updated_standard_catch <- bind_rows(combined_rst_wo_na_fl, na_filled_lifestage, no_catch) |> glimpse()
+
+# add include_in_model variable based on sampling window criteria
+# read in years to include produced in prep data for model
+gcs_get_object(object_name = "jpe-model-data/stream_week_site_year_include.csv",
+               bucket = gcs_get_global_bucket(),
+               saveToDisk = here::here("data", "model-data", "stream_week_site_year_include.csv"), overwrite = TRUE)
+
+years_to_include <- read_csv(here::here("data", "model-data", "stream_week_site_year_include.csv")) |> 
+  select(stream, site, monitoring_year, min_date, max_date) |> glimpse()
+
```

Comment:
I added >= or <= because if the date equals the min or max we want to include. I also added where we categorize as false all that are NA because if wasn't in list of streams to include then we should exclude.